share/sysbuild/cmake/modules/sysbuild_extensions.cmake

```diff
@@ -311,6 +308,11 @@ function(ExternalZephyrProject_Cmake)
       list(APPEND sysbuild_cache_strings "${cache_entry}\n")
     endif()
   endforeach()
+  if(DEFINED BOARD_REVISION)
+    list(APPEND sysbuild_cache_strings "BOARD:STRING=${BOARD}@${BOARD_REVISION}\n")
+  else()
+    list(APPEND sysbuild_cache_strings "BOARD:STRING=${BOARD}\n")
+  endif()
```

Comment:
This should be moved down and combined with:

```
if(${ZCMAKE_APPLICATION}_BOARD)
  ...
elseif(DEFINED BOARD_REVISION)
  ...
else()
  ...
endif()
```
Otherwise, the sysbuild cache may contain two BOARD values.

Reply:
> Otherwise, the sysbuild cache may contain two BOARD values.

No it will not.
It will contain a BOARD value which is the `GLOBAL` sysbuild value, and then this code:
https://github.com/zephyrproject-rtos/zephyr/blob/b8891f2fa5f4fba82dfe6222dbf8860213699be8/share/sysbuild/cmake/modules/sysbuild_extensions.cmake#L323-L327
will ensure a board entry in the form of `<image>_BOARD`, such as `mcuboot_BOARD` is added to the sysbuild image cache.

This ensures that this code:
https://github.com/zephyrproject-rtos/zephyr/blob/49c7aa56fa0c1e61de95577e5fdc1877872b549c/cmake/modules/extensions.cmake#L2555-L2560
will correctly pickup BOARD, if no dedicated `<image>_BOARD` is defined.

Reply:
as example, where a specific board is given for an image build:
```
$ cat build/blinky_sysbuild_cache.txt|grep BOARD
CACHED_BOARD:STRING=nrf9160dk_nrf9160
BOARD:STRING=nrf9160dk_nrf9160@0.7.0
blinky_BOARD:STRING=nrf52840dk_nrf52811

```